resources/js/components/UseExpandable.vue

```diff
@@ -0,0 +1,139 @@
+<script>
+
+import { ref, computed, onMounted } from 'vue'
+
+export default function useExpandable(itemSection, items, initialItem, setHash = true, wrap = true, vertical = false) {
+
+    const activeItem = ref('')
+    const currentFocusItem = ref('')
+
+    const itemKeys = computed(() => { return Object.keys(items)})
+    const firstItemKey = computed(() => { return itemKeys.value[0] })
+    const lastItemKey = computed(() => { return itemKeys.value[itemKeys.value.length - 1] })
+    const currentFocusItemIndex = computed(() => { return itemKeys.value.indexOf(currentFocusItem.value) })
+
+    const nextItemKey = computed(() => {
+        return currentFocusItem.value === lastItemKey.value
+            ? (wrap ? firstItemKey.value : null)
+            : itemKeys.value[currentFocusItemIndex.value + 1]
+    })
+
+    const prevItemKey = computed(() => {
+        return currentFocusItemIndex.value === 0
+            ? (wrap ? lastItemKey.value : null)
+            : itemKeys.value[currentFocusItemIndex.value - 1]
+    })
+
+    const navigateNextItem = (evt) => {
+        if (nextItemKey.value) {
+            evt.preventDefault();
+            const nextItemButton = itemSection.value.querySelector(`[id="${nextItemKey.value}"]`)
```

Comment:
Vue docs would recommend using `unref(itemSection)` everywhere instead of `itemSection.value` in case when `itemSection` is passed in it isn't a reference.